### PR TITLE
Support discarding open object versions.

### LINKED ISSUE #5194: Support discarding open versions (discard draft)
Currently, this is supported by RepositoryObject. This should be exposed via API and also include additional discard operations link cleaning up the workspace.

### PR DESCRIPTION
**Pull request by justinlittman**
closes #5194

## Why was this change made? 🤔
This functionality is needed for H3.

## How was this change tested? 🤨
Unit

⚡ ⚠ If this change has cross service impact, including data writes to shared file systems, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

### CODE REVIEW

File: app/models/repository_object.rb [hunk 1]
```diff
@@ -111,15 +111,21 @@ def close_version!(description: nil)
     end
   end
 
+  def check_discard_open_version!
```
Comment (lwrubel) [inline app/models/repository_object.rb:114]: Why is the check turned into a bang method? Just curious, not a blocker.

**Review by lwrubel — APPROVED** [review]

**Comment by lwrubel** [comment]
Approved although I am wondering if there any way to further test this in a deployed env?

**Comment by justinlittman** [comment]
I've tested it locally; it will get exercised by H3 eventually.

File: app/models/repository_object.rb (hunk 1 above)
Comment (justinlittman) [reply app/models/repository_object.rb:114]: Because public and it raises, it is a general convention.

**Pull request merged**